samples/userspace/hello_world_user/src/main.c

```diff
@@ -13,7 +13,9 @@ K_THREAD_STACK_DEFINE(user_stack, USER_STACKSIZE);
 
 static void user_function(void *p1, void *p2, void *p3)
 {
-	printf("Hello World from UserSpace! %s\n", CONFIG_BOARD);
+	printf("Hello World from %s (%s)\n",
+	       k_is_user_context() ? "UserSpace!" : "privileged mode.",
+	       CONFIG_BOARD);
```

Comment:
> This change turned out to be useful
> to me so I thought of contributing it to help the next person who might
> run into the same kind of issues.

In that case, should this be adding an assertion on `k_is_user_context` instead? After all, `user_function` is never expected to be invoked in privileged mode; and, if it ever is, then something is wrong.

```suggestion
	__ASSERT_NO_MSG(k_is_user_context());
	printf("Hello World from UserSpace! %s\n", CONFIG_BOARD);
```